src/tests/test_meta_blk_mgr.cpp

```diff
@@ -415,6 +420,31 @@ class VMetaBlkMgrTest : public ::testing::Test {
         }
     }
 
+    //
+    // 1. do a write, make sure compression is triggered.
+    // 2. update same meta blk, with data that exceeds compression ratio and thus back off compression.
+    //
+    void write_compression_backoff() {
+        HS_SETTINGS_FACTORY().modifiable_settings([](auto& s) {
+            s.metablk.compress_ratio_limit = 100; // this will allow every compression attempt;
+            HS_SETTINGS_FACTORY().save();
+        });
+
+        LOGINFO("compression ratio limit changed to: {}", m_mbm->get_compress_ratio_limit());
+
+        [[maybe_unused]] const auto write_result = do_sb_write(true /* do_overflow */, 15 * Mi);
```

Comment:
we removed the nodiscard from all apis, so this annoying maybe_unused assignment can be skipped and just do ```do_sb_write(...)```

Reply:
hmm... it yields build failure for me in master, maybe it is not the case in develop branch.

Reply:
/tmp/source/yk_develop/src/tests/test_meta_blk_mgr.cpp:435:20: error: ignoring return value of âuint64_t VMetaBlkMgrTest::do_sb_write(bool, size_t)â, declared with attribute ânodiscardâ [-Werror=unused-result]
  435 |         do_sb_write(true /* do_overflow */, 15 * Mi);
      |         ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~